### ISSUE TITLE
Missing X-rays???

### ISSUE THREAD

**Issue by ChiLyMan** [Bruker-Amics/bugs_features#82]
@MarkGuo7 I know you guys are looking for things to do so here is another interesting "feature" we've seen here.

This time the BSE is being collected, and the segmentation seems to have been done correctly, but this time some of the segments have not been x-rayed. If you look at the colored mineral map, those large grey areas near the lower half are all unknowns, but are unknown because no x-ray data was collected for them. As with the other issue, the BSE is not a problem, that is being collected correctly (see attached BSE image). And the particles are being segmented this time (see segmented image). Just this time it has decided not to collect x-rays for those points. The analysis conditions are the same as the "Missing Fields" issues, but I've attached the conditions below as well.

![Missing X-rays](https://user-images.githubusercontent.com/84643841/126699768-f30c66ea-1a02-4f95-aaff-75ee34b6c050.JPG)
![Missing X-rays (BSE)](https://user-images.githubusercontent.com/84643841/126699782-4ed7e9c4-2bc2-48f6-92f0-1895f6cde885.JPG)
![Missing X-rays (Segmented)](https://user-images.githubusercontent.com/84643841/126700048-e8df990c-910c-4a85-b11f-1c702be008cc.JPG)
![Missing X-rays (Conditions)](https://user-images.githubusercontent.com/84643841/126700184-5bd0fc98-9fda-4252-8ffa-29f14925382d.JPG)

**Comment by MarkGuo7** [comment]
Sorry Chi, never saw this issue before. We will collect all xrays generated from segmentation.
May I have this result? I can take a closer look. Thanks

**Comment by ChiLyMan** [comment]
Hey Mark,

Sorry didn’t answer this question, but yes can send you the data for this sample once I get back home tonight.

Chi

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by ChiLyMan** [comment]
Morning/Evening Mark,

Please find the zipped up datafile for that sample with no x-rays. Let me or Nigel know if you need more data, as this is the only file I have, Nigel will need to get you any info from the machine.

[Ma5002_3842_Run_1.zip](https://github.com/Bruker-Amics/bugs_features/files/6879378/Ma5002_3842_Run_1.zip)

**Comment by ChiLyMan** [comment]
@MarkGuo7 
Here is another example of the missing x-rays. Looks like it has affected the whole field this time. Settings are the same as the other data sets. I've included the data set for the same for you. Again if you need any additional data from this data Nigel can get it from the machine. 

Note I believe Nigel has updated the software on the system to 3.1.1.2444. I believe some of the examples I sent before might have been from an older version (Nigel can you confirm?)

[Ma5002_3930.zip](https://github.com/Bruker-Amics/bugs_features/files/6927256/Ma5002_3930.zip)
![image](https://user-images.githubusercontent.com/84643841/128091681-9fd0e261-863f-4231-9cd2-50c908045b9f.png)
![image](https://user-images.githubusercontent.com/84643841/128091710-d5602431-c311-4bb1-b153-6d5d79d196dd.png)

**Comment by bernie1034** [comment]
Correct - the measurements with the issue first occurring were from an earlier version (v2084? Will have to check). All measurements were re-run using v2444. Check the version in the measurement log, to double-check.

**Comment by bernie1034** [comment]
I just opened that measurement on my machine, and it seems this field is ok - there are X-rays collected:

![image](https://user-images.githubusercontent.com/84200879/128098289-f1b3cd82-2f34-418b-98e0-85f6a2147467.png)

**Comment by ChiLyMan** [comment]
OK let me try again.

**Comment by ChiLyMan** [comment]
Hahaha...now when I click on the x-ray spot process just crashes. I tried reclassifying the dataset, and that didn't fix the greyed out field and clicking the x-ray spot still crashes process. I'll fiddle some more and give you feedback....

**Comment by ChiLyMan** [comment]
@bernie1034  Not sure how you got it to read the x-ray but this is what I've tried:-
* Reclassify the dataset
* Open original one Nigel set me before any processing.
* Downloaded the one I uploaded (notice the MRE file was missing so just used the .cache version)
* Updated my software to v3.1.1.2495

And in all cases when I load the processed, reprocessed, unprocessed, uploaded version, 1) that area is listed as unknown and when I try to click on the x-ray point process crashes. Only thing I've not tried in your internal beta version...

let me know if you want me to try anything else.

**Comment by MarkGuo7** [comment]
@ChiLyMan Hi Chi, thanks for these reporting.
For some reason, the xray data didn't save properly. The issue field id is no.6, and you can see the size of its xray file is 0.

![image](https://user-images.githubusercontent.com/84481724/128104492-bc74b45c-3c7b-4802-b2f4-37bab50c3b38.png)

From the Investigator log, seems the saving is normal.

![image](https://user-images.githubusercontent.com/84481724/128104852-5f185f1e-76ca-40cb-97e3-dc6701634265.png)

May I know what's type of the disk you are using? Not sure why it didn't save properly.

That's why you couldn't see its xrays and cause crashes when you click it.
I will fix the crashes issue, but need to find out why it didn't save properly.

**Comment by MarkGuo7** [comment]
@ChiLyMan Let me know if you want to have a meeting. :)

**Comment by ChiLyMan** [comment]
@MarkGuo7 Morning Mark. No need if you've figured out it a file issue. Need to ask Nigel to check the original file on his end to see if the file on the instrument PC contains the x-ray data. Then it becomes a stupid Windows playing funny bugger with us issue.

**Comment by ChiLyMan** [comment]
Just so we have it documented here, Nigel and I think the issue is not a writing issue of AMICS, because the  xray file on the system was ok. But seems to be an access/copy issue from windows. I believe NIgel is going to do some more testing because the original example I posted, the x-ray file I copied is only partial, which is why half the field has no quantification (and also Process does not crash in that case).
